kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/arbs.kt

```diff
@@ -43,18 +50,26 @@ fun <A> Arb<A>.filterNot(f: (A) -> Boolean): Arb<A> = filter { !f(it) }
 fun <A, B> Arb<A>.map(f: (A) -> B): Arb<B> = object : Arb<B>() {
    override fun edgecases(): List<B> = this@map.edgecases().map(f)
 
+   override fun generateEdgecase(rs: RandomSource): B = f(this@map.generateEdgecase(rs))
+
    override fun values(rs: RandomSource): Sequence<Sample<B>> {
       return this@map.values(rs).map { Sample(f(it.value), it.shrinks.map(f)) }
    }
 
-   override fun sample(rs: RandomSource): Sample<B> = Sample(f(this@map.sample(rs).value))
+   override fun sample(rs: RandomSource): Sample<B> =
+      this@map.sample(rs).let {
+         Sample(f(it.value), it.shrinks.map(f))
```

Comment:
that looked like a bug, map previously discarded the shrinker

Reply:
nice spot.